### ISSUE TITLE
Error/Crash in editor

### ISSUE THREAD

**Issue by Craig-Macomber** [RileyA/STATICly-MAD#24]
You get this if you click something that has an action marker on it:

ArgumentError: Error #2025: The supplied DisplayObject must be a child of the caller.
    at flash.display::DisplayObjectContainer/removeChild()
    at Player/update()
    at Level/update()
    at Editor::EditorState/update()
    at Game/update()
    at Main/update()

**Comment by Hallight** [comment]
Yea I don't know where it's getting called, but it's from trying to remove a sprite child when that sprite isn't actually a child of the sprite youre removing it from.

**Comment by RileyA** [comment]
Basically the editor deletes and recreates a block whenever any of their properties change (and if you drag and then drop a block (even if you don't actually move it anywhere), its position changes and it gets destroyed and recreated, this is kinda sub-optimal, but it works well enough (earlier I made it just reposition if all you did was move it, but tracks and stuff didn't behave)).

So when you click (and release, it only rebuilds when you drop) the block is killed (along with its children), then when Player::update tries to remove the child from the newly created block it isn't there and errors ensue.

I just fixed this by explicitly checking if it the sprite has the marker as a child before attempting to remove it.
